{"bug_class": "PROVER", "kind": "ERROR", "bug_type": "RESOURCE_LEAK", "qualifier": "resource of type `java.io.FileWriter` acquired to `writer` by call to `new()` at line 142 is not released after line 147.\n**Note**: potential exception at line 143", "severity": "HIGH", "visibility": "user", "line": 147, "column": -1, "procedure": "void PerfTest.reportResult(String,String,Histogram)", "procedure_id": "org.apache.logging.log4j.core.async.perftest.PerfTest.reportResult(java.lang.String,java.lang.String.f739607919908d7d3cd95d9e82a639d6", "procedure_start_line": 136, "file": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "bug_trace": [{"level": 0, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 136, "column_number": -1, "description": "start of procedure reportResult(...)"}, {"level": 0, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 138, "column_number": -1, "description": ""}, {"level": 1, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 161, "column_number": -1, "description": "start of procedure createSamplingReport(...)"}, {"level": 1, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 162, "column_number": -1, "description": ""}, {"level": 1, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 163, "column_number": -1, "description": "Taking true branch"}, {"level": 1, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 164, "column_number": -1, "description": ""}, {"level": 1, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 172, "column_number": -1, "description": "return from a call to String PerfTest.createSamplingReport(String,Histogram)"}, {"level": 0, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 139, "column_number": -1, "description": ""}, {"level": 1, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 155, "column_number": -1, "description": "start of procedure println(...)"}, {"level": 1, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 156, "column_number": -1, "description": "Taking true branch"}, {"level": 1, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 157, "column_number": -1, "description": "Skipping println(...): unknown method"}, {"level": 1, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 159, "column_number": -1, "description": "return from a call to void PerfTest.println(String)"}, {"level": 0, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 141, "column_number": -1, "description": "Taking true branch"}, {"level": 0, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 142, "column_number": -1, "description": "Skipping FileWriter(...): unknown method"}, {"level": 0, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 143, "column_number": -1, "description": "exception java.io.IOException"}, {"level": 0, "filename": "log4j-core/src/test/java/org/apache/logging/log4j/core/async/perftest/PerfTest.java", "line_number": 147, "column_number": -1, "description": ""}], "key": "PerfTest.java|reportResult|RESOURCE_LEAK", "node_key": "48815eab4c26cee962db00712b67bbd0", "hash": "c2746a17a9c895c8c163640cf010aa43", "bug_type_hum": "Resource Leak", "censored_reason": ""}